src/Stormpath/DataStore/DefaultDataStore.php

```diff
@@ -144,6 +163,7 @@ public function save(Resource $resource, $returnType = null)
 
     public function delete(Resource $resource)
     {
+        $this->deleteCachableResource($resource);
```

Comment:
This should happen after executing the delete request on the server so we know it succeeded.